### PR TITLE
model 1.1 implementation

### PR DESCRIPTION
**Pull request by hussain-jafari**
### Description]
- *Category*: implementation
- *JIRA issue*: [MIC-4397](https://jira.ihme.washington.edu/browse/MIC-4397)
- *Research reference*: [gestational age](https://vivarium-research.readthedocs.io/en/latest/models/intervention_models/maternal_supplementation/index.html#gestational-age)

### Changes and notes
Add effect of IFA and MMS supplementation on gestational age exposure. 
Initialize raw gestational age exposure in LBWSG component. 
Deduplicate effect pipeline names in additive risk effects with same risk by including target in name and update names in BirthweightShift component to reflect this. 

### Verification and Testing
Initialized 3 fake populations with simulants on every type of supplementation: one without supplementation, one with just IFA, and one with MMS and IFA. Checked that simulants on each type of supplementation had their gestational age change as expected in each scenario.

### CODE REVIEW

File: src/vivarium_gates_nutrition_optimization_child/components/maternal_characteristics.py [hunk 1]
```diff
@@ -272,11 +277,115 @@ def get_effect(self, index: pd.Index) -> pd.Series:
...
+    def _get_mms_excess_shift_data(
+        self, builder: Builder, key: str
+    ) -> Dict[str, LookupTable]:
+        excess_shift_data = builder.data.load(
+            key, affected_entity=self.target.name, affected_measure=self.target.measure
+        )
+        excess_shift_data = self.build_excess_shift_lookup_table(builder, excess_shift_data)
+        return excess_shift_data
+
+    def build_excess_shift_lookup_table(
```
Comment (rmudambi) [inline src/vivarium_gates_nutrition_optimization_child/components/maternal_characteristics.py:327]: Why did you need to extract this to a separate method? Seems like it could just have been part of `_get_mms_excess_shift_data`

File: src/vivarium_gates_nutrition_optimization_child/components/maternal_characteristics.py [hunk 2]
```diff
@@ -272,11 +277,115 @@ def get_effect(self, index: pd.Index) -> pd.Series:
...
+    ##################################
+    def _get_excess_shift_source(self, builder: Builder) -> Pipeline:
+        return builder.value.register_value_producer(
+            self.excess_shift_pipeline_name,
+            source=self.get_excess_shift,
+            requires_columns=[self.raw_gestational_age_exposure_column_name],
+        )
+
+    def _get_risk_specific_shift_source(self, builder: Builder) -> Pipeline:
+        return builder.value.register_value_producer(
```
Comment (rmudambi) [inline src/vivarium_gates_nutrition_optimization_child/components/maternal_characteristics.py:348]: Do we have any modifiers on this pipeline? If not, why can't it just be a LookupTable which has 0 as its source?

File: src/vivarium_gates_nutrition_optimization_child/data/raw/ifa_gestational_age_shifts/nigeria.csv
Comment (rmudambi) [inline src/vivarium_gates_nutrition_optimization_child/data/raw/ifa_gestational_age_shifts/nigeria.csv:1]: These shouldn't be here. Your PR is pointed at `main` when it should be pointed at `feature/MIC-4407-model_1.1_artifact`

**Review by rmudambi — APPROVED** [review]

File: src/vivarium_gates_nutrition_optimization_child/components/maternal_characteristics.py (hunk 1 above)
Comment (hussain-jafari) [reply src/vivarium_gates_nutrition_optimization_child/components/maternal_characteristics.py:327]: Oh yeah it was in a separate method because I used it multiple times in my old implementation but now it's only called once so I'll move it in.

File: src/vivarium_gates_nutrition_optimization_child/data/raw/ifa_gestational_age_shifts/nigeria.csv
Comment (hussain-jafari) [reply src/vivarium_gates_nutrition_optimization_child/data/raw/ifa_gestational_age_shifts/nigeria.csv:1]: This is because I started with implementation and later realized it would be useful to have data in the artifact so some of implementation was already written when I started writing the artifact code. They work independently so I based both off of main and was going to merge the artifact branch into main and rebase off of that when merging implementation.

File: src/vivarium_gates_nutrition_optimization_child/components/maternal_characteristics.py [hunk 3]
```diff
@@ -272,11 +277,101 @@ def get_effect(self, index: pd.Index) -> pd.Series:
...
+    # Pipeline sources and modifiers #
+    ##################################
+    def _get_excess_shift_source(self, builder: Builder) -> Pipeline:
+        return builder.value.register_value_producer(
+            self.excess_shift_pipeline_name,
+            source=self.get_excess_shift,
+            requires_columns=[self.raw_gestational_age_exposure_column_name],
+        )
+
+    def _get_risk_specific_shift_source(self, builder: Builder) -> Pipeline:
```
Comment (rmudambi) [inline src/vivarium_gates_nutrition_optimization_child/components/maternal_characteristics.py:340]: This is typed incorrectly now.

**Review by rmudambi — APPROVED** [review]
